test/framework/config.go

```diff
@@ -0,0 +1,185 @@
+	DefaultGatewayNamespace       string            `yaml:"gatewayNamespace,omitempty"`
+	KumaCPImageRepo               string            `yaml:"cpImageRepo,omitempty" envconfig:"KUMA_CP_IMAGE_REPOSITORY"`
+	KumaDPImageRepo               string            `yaml:"dpImageRepo,omitempty" envconfig:"KUMA_DP_IMAGE_REPOSITORY"`
+	KumaInitImageRepo             string            `yaml:"initImageRepo,omitempty" envconfig:"KUMA_INIT_IMAGE_REPOSITORY"`
+	KumaUniversalImageRepo        string            `yaml:"universalImageRepo,omitempty"`
+	XDSApiVersion                 string            `yaml:"xdsVersion,omitempty" envconfig:"API_VERSION"`
+	K8sType                       K8sType           `yaml:"k8sType,omitempty" envconfig:"KUMA_K8S_TYPE"`
+	IPV6                          bool              `yaml:"ipv6,omitempty" envconfig:"IPV6"`
+	UseHostnameInsteadOfIP        bool              `yaml:"useHostnameInsteadOfIP,omitempty" envconfig:"KUMA_USE_HOSTNAME_INSTEAD_OF_ID"`
+	UseLoadBalancer               bool              `yaml:"useLoadBalancer,omitempty" envconfig:"KUMA_USE_LOAD_BALANCER"`
+	CIDR                          string            `yaml:"kumaCidr,omitempty"`
+	DefaultClusterStartupRetries  int               `yaml:"defaultClusterStartupRetries,omitempty" envconfig:"KUMA_DEFAULT_RETRIES"`
+	DefaultClusterStartupTimeout  time.Duration     `yaml:"defaultClusterStartupTimeout,omitempty" envconfig:"KUMA_DEFAULT_TIMEOUT"`
+	KumactlBin                    string            `yaml:"kumactlBin,omitempty" envconfig:"KUMACTLBIN"`
+
+	SuiteConfig SuiteConfig `yaml:"suites,omitempty"`
+}
+
+type SuiteConfig struct {
+	Compatibility CompatibilitySuiteConfig `yaml:"compatibility,omitempty"`
+	Helm          HelmSuiteConfig          `yaml:"helm,omitempty"`
+}
+
+type CompatibilitySuiteConfig struct {
+	HelmVersion string `yaml:"helmVersion,omitempty"`
+}
+
+type HelmSuiteConfig struct {
+	Versions      []string `yaml:"versions,omitempty"`
+	ExtraYamlPath string   `yaml:"extraYaml,omitempty"`
+}
+
+func (c E2eConfig) Sanitize() {
+}
+
+func (c E2eConfig) Validate() error {
+	if Config.CNIConf.ConfName == "" {
+		switch Config.K8sType {
+		case KindK8sType:
+			Config.CNIConf = CniConf{
+				ConfName: "10-kindnet.conflist",
+				NetDir:   "/etc/cni/net.d",
+				BinDir:   "/opt/cni/bin",
+			}
+		case K3dK8sType:
+			Config.CNIConf = CniConf{
+				ConfName: "10-flannel.conflist",
+				NetDir:   "/var/lib/rancher/k3s/agent/etc/cni/net.d",
+				BinDir:   "/bin",
+			}
+		case AzureK8sType:
+			Config.CNIConf = CniConf{
+				ConfName: "10-azure.conflist",
+				NetDir:   "/etc/cni/net.d",
+				BinDir:   "/opt/cni/bin",
+			}
+		case AwsK8sType:
+			Config.CNIConf = CniConf{
+				ConfName: "10-aws.conflist",
+				NetDir:   "/etc/cni/net.d",
+				BinDir:   "/opt/cni/bin",
+			}
+		default:
+			return fmt.Errorf("you must set a supported KUMA_K8S_TYPE got:%s", Config.K8sType)
+		}
+	}
+	if Config.IPV6 && Config.CIDR == "" {
+		Config.CIDR = "fd00:fd00::/64"
+	}
```

Comment:
nit: the above is more of `autoconfiguration` than validation. I think I'd move this to `func autoconfigure(c *E2eConfig)` to follow the same pattern as with production config. And call this after `core_config.Load`